.github/workflows/integration_test.yml

```diff
@@ -7,16 +7,40 @@ on:
 jobs:
   run:
     name: Integration Test
-    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: [amd64, arm64]
+    runs-on: ${{ fromJson('{"amd64":"ubuntu-latest", "arm64":["self-hosted", "Linux", "ARM64"]}')[matrix.arch] }}
     steps:
       - uses: actions/checkout@master
         with:
           # Must use at least depth 2!
           fetch-depth: 2
 
       - name: Create kind cluster
-        uses: helm/kind-action@v1.2.0
+        # released version of kind-action doesn't support arm64
+        uses: helm/kind-action@4c7909140acfc81a05fc96fed8fea6673ba8ce80
+        with:
+          kubectl_version: v1.23.1
+
+      - name: Setup Helm
+        uses: azure/setup-helm@v2.1
+
+      - name: Setup Kubectl
+        uses: azure/setup-kubectl@v2.0
+
+      - name: Magic Kind DNS Fix
+        if: ${{ matrix.arch == 'arm64' }}
+        run: |
+          docker exec chart-testing-control-plane /bin/bash -c "sed -e 's/nameserver \(.*\)/nameserver 8.8.8.8/g' /etc/resolv.conf > /etc/resolv.conf.sed"
```

Comment:
The network problem in kind is all about DNS.

`kind` has some magic scripts to fix the DNS loop in coredns when running inside a docker, see https://github.com/kubernetes-sigs/kind/blob/main/images/base/files/usr/local/bin/entrypoint#L375, and so does `minikube`. The awful thing is that: these scripts run, iptables rules are also applied successfully, but the DNS just don't work.

I have entered the environment manually, and dig/nslookup/tcpdump/... the DNS request with a lot of tools, but both the redirection source/target doesn't respond the DNS request. To further understand the problem, we could trace with the iptables, enable debug log of dockerd... It will take a lot more time to find the root, so I choose a much simpler way to solve: replace the dns server to 8.8.8.8 ...